ivadomed/inference.py

```diff
@@ -324,6 +307,47 @@ def split_classes(nib_prediction):
         pred_list.append(class_pred)
     return pred_list
 
+def reconstruct_3d_object(context, batch, undo_transforms, preds, preds_list, kernel_3D, volume, 
```

Comment:
I recommend future PR to start thinking of reducing the parameter complexities, either via setting classes or configuration modules/dict etc to avoid having large number of parameters past around such as shown. I can work on those PRs drawing some inspirations from Detectron2's approach but I believe everything in this PR is a very important first step before we worry about those. The original segmentation function was way complex for trying to do too many things in a single function that can be further streamlined after this PR.

Reply:
I agree with everything mentioned here! I think given the large number of changes needed in reducing parameter complexities it would have been best placed in another PR. Taking advantage of class settings would definitely help with this process